TypeSystem/Schema/schema.h

```diff
@@ -390,16 +394,26 @@ struct LanguageSyntaxCPP : CurrentStructPrinter<CPP_LANGUAGE_SELECTOR> {
         os_ << "CURRENT_NAMESPACE(" << nmspc << ") {\n";
         for (const auto& input_type : types_) {
           const auto& type_substance = input_type.second;
-          if (Exists<ReflectedType_Struct>(type_substance) || Exists<ReflectedType_Variant>(type_substance) ||
-              Exists<ReflectedType_Enum>(type_substance)) {
+          // Do not expose types that are templated, as their names may collide.
+          if ((Exists<ReflectedType_Struct>(type_substance) &&
+               !Exists(Value<ReflectedType_Struct>(type_substance).template_id)) ||
+              Exists<ReflectedType_Variant>(type_substance) || Exists<ReflectedType_Enum>(type_substance)) {
             os_ << "  CURRENT_NAMESPACE_TYPE(" << TypeName(input_type.first, "::")
                 << ", current_userspace::" << TypeName(input_type.first) << ");\n";
+          } else if (Exists<ReflectedType_Struct>(type_substance) &&
+                     Exists(Value<ReflectedType_Struct>(type_substance).template_id)) {
+            const std::string type_name = TypeName(input_type.first, "::");
+            os_ << "  CURRENT_NAMESPACE_TYPE("
+                << Value<ReflectedType_Struct>(type_substance).TemplateFullNameDecorator()(type_name)
+                << ", current_userspace::" << TypeName(input_type.first) << ");\n";
           }
         }
         if (Exists(namespace_to_expose_)) {
           const NamespaceToExpose& expose = Value(namespace_to_expose_);
           os_ << "\n  // Privileged types.\n";
           for (const auto& t : expose.types) {
+            // NOTE(dkorolev): If the uses requests to expose a templated type, it's their responsibility ]
+            // to only do it for at most one instantiation per type.
```

Comment:
So, templated can be with only one template typename? Then maybe it's easier to drop support for the templated types and force to use explicit compile-time Variants?